cmd/jujud/agent/machine.go

```diff
@@ -1191,6 +1192,68 @@ func (a *MachineAgent) startEnvWorkers(
 	return runner, nil
 }
 
+// startDyingEnvWorkers starts state server workers that need to run per
+// dying environment.
+func (a *MachineAgent) dyingEnvWorkers(
+	ssSt envworkermanager.InitialState,
+	st *state.State,
+) (_ worker.Worker, err error) {
+	envUUID := st.EnvironUUID()
```

Comment:
Given how similar the top of dyingEnvWorkers and startEnvWorkers, please consider extracting and sharing the common bits.